src/Shared/Recruit.Vacancies.Client/Application/Aspects/RequestPerformanceBehaviour.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.Logging;
+
+namespace Esfa.Recruit.Vacancies.Client.Application.Aspects
+{
+    public class RequestPerformanceBehaviour<TRequest, TResponse> : IPipelineBehavior<TRequest, TResponse>
+    {
+        private readonly Stopwatch _timer;
+        private readonly ILogger<TRequest> _logger;
+
+        public RequestPerformanceBehaviour(ILogger<TRequest> logger)
+        {
+            _timer = new Stopwatch();
+
+            _logger = logger;
+        }
+
+        public async Task<TResponse> Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate<TResponse> next)
+        {
+            _timer.Start();
+
+            var response = await next();
```

Comment:
What happens if the next handler throws an exception?  Do we need to wrap this in a try/catch ?

Reply:
I am presuming it will bubble up the error and we don't know how to treat the exception here.

Reply:
But we'd still want to write out the log though.  Catch, write the log, rethrow.

Reply:
The end game is this we want to determine if a particular request is slow performing, which means we only want to log the successful calls.